app/screens/post_options/options/save_option/save_option.tsx

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+
+import {deleteSavedPost, savePostPreference} from '@actions/remote/preference';
+import {Screens} from '@constants';
+import {useServerUrl} from '@context/server';
+import {t} from '@i18n';
+import {dismissBottomSheet} from '@screens/navigation';
+
+import BaseOption from '../base_option';
+
+type CopyTextProps = {
+    isSaved: boolean;
+    postId: string;
+}
+
+const SaveOption = ({isSaved, postId}: CopyTextProps) => {
+    const serverUrl = useServerUrl();
+
+    const onHandlePress = useCallback(async () => {
+        const remoteAction = isSaved ? deleteSavedPost : savePostPreference;
+        await remoteAction(serverUrl, postId);
```

Comment:
(Probably out of the scope of this PR) This probably is fast, but what if it is not (bad connection to the server)? The bottom sheet just stays there without any feedback (and potentially you hit the button a few more times) until the message gets through?
Shouldn't we put some "loading state"? (maybe only if it is taking too long)

Reply:
maybe we just remove the await on the action..

Reply:
done